earthfile2llb/shell.go

```diff
@@ -76,11 +76,29 @@ func strWithEnvVarsAndDocker(args []string, envVars []string, withShell, withDeb
 
 type shellWrapFun func(args []string, envVars []string, withShell, withDebugger, forceDebugger bool) []string
 
+func checkEnvPresent(envSlice []string) bool {
+	for _, element := range envSlice {
+		if element == "env" {
+			return true
+		}
+	}
+	return false
+}
 func withShellAndEnvVars(args []string, envVars []string, withShell, withDebugger, forceDebugger bool) []string {
-	return []string{
-		"/bin/sh", "-c",
-		strWithEnvVarsAndDocker(args, envVars, withShell, withDebugger, forceDebugger, false, false, "", ""),
+	envSlice := []string{"/bin/sh", "-c"}
+
+	for _, env := range envVars {
+		parts := strings.SplitN(env, "=", 2)
+		if parts[0][0] == '\'' {
```

Comment:
this feels odd; but perhaps it will work?

can you update the test to include this:

```
FROM alpine
ENV mid'quote' yes
RUN env | grep "midquote=yes"
```

to test if this code works.

Reply:
if this fix does work, I think it's more appropriate to move this fix into strWithEnvVarsAndDocker

Reply:
we get `mid'quote'=yes` this value in env

Reply:
fixed

Reply:
since we are replacing `envVars` in the fix, it's better to keep it at same function as `strWithEnvVarsAndDocker` is called from multiple functions.